vue/mixins/logic.js

```diff
@@ -1,6 +1,340 @@
+            return {
+                brand: this.brand,
+                model: this.model,
+                version: this.version,
+                currency: this.currency
+            };
+        },
+        configOptionsStructure() {
+            return {
+                structure: this.structure,
+                currency: this.currency
+            };
+        }
+    },
+    watch: {
+        configOptions: {
+            handler: async function(value, previous) {
+                if (!this.ripeData || !this.configData) return;
+
+                // resets the parts and personalization options if
+                // the model was changed but they stayed the same,
+                // which makes them invalid
+                if (this.shouldReset(value, previous)) {
+                    value.parts = null;
+                    value.initials = "";
+                    value.engraving = null;
+                    value.initialsExtra = {};
+                }
+
+                // makes the configuration call with only the changed
+                // values, defaulting to the 'data' values in the unchanged
+                // ones, safeguarding against outdated props
+                await this.configRipe({
+                    brand: value.brand !== previous.brand ? value.brand : undefined,
+                    model: value.model !== previous.model ? value.model : undefined,
+                    version: value.version !== previous.version ? value.version : undefined,
+                    currency: value.currency !== previous.currency ? value.currency : undefined,
+                    parts: value.parts,
+                    initials: value.initials !== previous.initials ? value.initials : undefined,
+                    engraving: value.engraving !== previous.engraving ? value.engraving : undefined,
+                    initialsExtra: !this.equalInitialsExtra(
+                        value.initialsExtra,
+                        previous.initialsExtra
+                    )
+                        ? value.initialsExtra
+                        : undefined
+                });
+            },
+            deep: true
+        },
+        configOptionsStructure: {
+            handler: async function(value, previous) {
+                if (!this.ripeData || !this.configData) return;
+
+                // verifies if there were no changes in the structure
+                // and/or currency comparing to the previous structure
+                // prop and the structureData, making the configuration
+                // call with only the changed values and defaulting to
+                // the 'data' values for the others, which are updated
+                const structure = value.structure;
+                const previousStructure = previous.structure;
+                const unchangedStructure =
+                    this.equalStructure(structure, this.structureData) ||
+                    this.equalStructure(structure, previousStructure);

```

Comment:
Why is this and/or only a problem with structure? The rest of the "has this changed" verifications don't look at data

Reply:
The problem was due to the fact that the `_copyRipeData` function called by the events of initials, parts and others were causing the `update:structure` event to be sent, triggering the change in structure and causing the `configRipe` to be called several times.